### ISSUE TITLE
implement `Result` struct

### ISSUE THREAD

**Issue by cwinters8** [cwinters8/gomap#1]
Using `Invocation` for both making method calls and parsing responses is not working well. The `methodResults` object returned with a response seems to often have different fields than are necessary for a request method.

Right now, this is causing the [`TestMailbox/draft` test](https://github.com/cwinters8/gomap/blob/4c8170245ed3f4c254ac068d6733cf670c2c6c04/mailbox_test.go#L29-L56) to fail with the error:

```error
mailbox_test.go:48: failed to create draft email: failed to send email set request: failed to unmarshal response body: failed to unmarshal args: failed to coerce create value to map. value is nil
```

because `create` is not an attribute on the result of the Email/set call. Instead, it looks something like

```go
map[string]interface {} [
	"destroyed": nil,
	"notCreated": nil,
	"notDestroyed": nil,
	"notUpdated": nil,
	"accountId": "u69394015",
	"created": map[string]interface {} [
		"fb0c675a-01e4-4764-8677-bd6a1a3ecb6b": *(*interface {})(0xc0002474c8),
	],
	"newState": "3566",
	"oldState": "3564",
	"updated": nil,
]
```

Currently thinking that a `Result` struct that could either:

- grow to include many fields to support different methods
- use generics to make Result have only the necessary fields/values for a given method

The latter would be cleaner, but implementing will be more complex.

**Comment by cwinters8** [comment]
Attempting to work out a way to parse a response into a certain type of result struct based on the type of arguments provided.

**Comment by cwinters8** [comment]
Result and Results structs are created and tests indicate correct functionality. Next step is to utilize them when parsing request responses.

**Comment by cwinters8** [comment]
Resolved with commit ac60a714e71e90f10171eddb0f09a840f53afc72
